android/sdl_android/src/main/java/com/smartdevicelink/util/SdlAppInfo.java

```diff
@@ -181,6 +181,40 @@ public String toString() {
         }
         return vehicleMakesList;
     }
+
+    static boolean checkIfVehicleSupported(List<VehicleType> supportedVehicleList, VehicleType connectedVehicle){
+        if (null == supportedVehicleList || null == connectedVehicle){
+            return false;
+        }
+        if (supportedVehicleList.contains(connectedVehicle)){
+            return true;
+        }
+        for (VehicleType supportedVehicle :supportedVehicleList) {
+            String svMake = supportedVehicle.getMake();
+            String cvMake = connectedVehicle.getMake();
+            if(svMake != null && cvMake != null && cvMake.equalsIgnoreCase(svMake)) {
+                String svModel = supportedVehicle.getModel();
+                String cvModel = connectedVehicle.getModel();
+                if (svModel != null && cvModel != null && cvModel.equalsIgnoreCase(svModel)) {
+                    boolean ret = true;
+                    String svMY = supportedVehicle.getModelYear();
+                    String cvMY = connectedVehicle.getModelYear();
+                    if (svMY != null && cvMY != null) {
+                        ret = cvMY.equalsIgnoreCase(svMY);
+                    }
+                    String svTrim = supportedVehicle.getTrim();
+                    String cvTrim = connectedVehicle.getTrim();
+                    if (svTrim != null && cvTrim != null) {
+                        ret &= cvTrim.equalsIgnoreCase(svTrim);
+                    }
+                    if (ret){
+                        return true;
```

Comment:
@kostyaBoss On line 211 you are returning from inside a `for` loop.  Can you change your logic so you return outside of the `for` loop?

Reply:
@santhanamk the logic is correct I'd describe it like "the first among all" so there is no need to finish the loop once one is found